js/src/ui/MethodDecoding/methodDecodingStore.js

```diff
@@ -173,6 +181,68 @@ export default class MethodDecodingStore {
       });
   }
 
+  decodeContractCreation (data, contractAddress) {
+    const result = {
+      ...data,
+      contract: true,
+      deploy: true
+    };
+
+    const { input } = data;
+    const abi = this._contractsAbi[contractAddress];
+
+    if (!input || !abi || !abi.constructors || abi.constructors.length === 0) {
+      return Promise.resolve(result);
+    }
+
+    const constructorAbi = abi.constructors[0];
+
+    const rawInput = /^(?:0x)?(.*)$/.exec(input)[1];
+
+    return this
+      .getCode(contractAddress)
+      .then((code) => {
+        if (!code || /^(0x)0*?$/.test(code)) {
+          return result;
+        }
+
+        const rawCode = /^(?:0x)?(.*)$/.exec(code)[1];
+        const codeOffset = rawInput.indexOf(rawCode);
+
+        if (codeOffset === -1) {
+          return result;
+        }
+
+        // Params are the last bytes of the transaction Input
+        // (minus the bytecode). It seems that they are repeated
+        // twice
+        const params = rawInput.slice(codeOffset + rawCode.length);
+        const paramsBis = params.slice(params.length / 2);
```

Comment:
So the thing is that from what I saw, the transaction input is : `somehting` . `bytecode` . `constructor_params` . `constructor_params`
Ie. the constructor params are concatenated twice at the end of the input. From what I could find online (which isn't much), it seems that the parameters should appear once. That's why I thought we could try the two cases, and result with which ever is correct.